test/realm_test.dart

```diff
@@ -56,6 +56,25 @@ class _Team {
   late final List<int> scores;
 }
 
+@RealmModel()
+class _Student {
+  @PrimaryKey()
+  late final int number;
+  late String? name;
+  late int? yearOfBirth;
+  late _School? school;
+}
+
+@RealmModel()
+class _School {
+  @PrimaryKey()
+  late final String name;
+  late String? city;
+  final List<_Student> students = [];
```

Comment:
I suggest we can implement the generator to insist the lists to be define as **final** without **late** and to be initialized **[]**. This is how the model is convenient to be used, because we will have compile time error that setter of the list is not able to be used. Otherwise they can use it, but the data are not stored in realm.
If the generator insists declaration of the lists to be like this 
`final List<_Student> students = [];`
 then this mistake 
`school.students = [student1, student2]`
will not be allowed and early visible.